enterprise/api/v1/views/enterprise_group.py

```diff
@@ -42,17 +48,44 @@ def get_queryset(self, **kwargs):
                 associated_customers.append(user_object.enterprise_customer)
             queryset = queryset.filter(enterprise_customer__in=associated_customers)
 
-        if self.request.method in ('GET',):
-            if learner_uuids := self.request.query_params.getlist('learner_uuids'):
+        if self.request.method == 'GET':
+            if learner_ids := self.request.query_params.getlist('learner_ids'):
                 # groups can apply to both existing and pending users
                 queryset = queryset.filter(
-                    Q(members__enterprise_customer_user__in=learner_uuids) |
-                    Q(members__pending_enterprise_customer_user__in=learner_uuids),
+                    Q(members__enterprise_customer_user__in=learner_ids) |
+                    Q(members__pending_enterprise_customer_user__in=learner_ids),
                 )
             if enterprise_uuids := self.request.query_params.getlist('enterprise_uuids'):
                 queryset = queryset.filter(enterprise_customer__in=enterprise_uuids)
         return queryset
 
+    @permission_required(
+        'enterprise.can_access_admin_dashboard',
+        fn=lambda request, *args, **kwargs: get_enterprise_customer_from_enterprise_group_id(kwargs['pk'])
+    )
+    def update(self, request, *args, **kwargs):
+        """
+        PATCH /enterprise/api/v1/enterprise-group/<group uuid>
+        """
+        if requested_customer := self.request.data.get('enterprise_customer'):
+            # Essentially checking ``enterprise.can_access_admin_dashboard`` but for the customer the requester is
+            # attempting to update the group record to.
+            implicit_access = rules.has_implicit_access_to_dashboard(self.request.user, requested_customer)
+            explicit_access = rules.has_explicit_access_to_dashboard(self.request.user, requested_customer)
+            if not implicit_access and not explicit_access:
+                return Response('Unauthorized', status=401)
+        return super().update(request, *args, **kwargs)
```

Comment:
Currently if `enterprise_customer` is blank it looks like access checks will be skipped.  We should probably default to blocking access in that case.
```suggestion
            if not implicit_access and not explicit_access:
                return Response('Unauthorized', status=401)
            return super().update(request, *args, **kwargs)
        return Response('Unauthorized', status=401)
```